src/Meshes/mesh.ts

```diff
@@ -1302,18 +1302,49 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
      * This method recomputes and sets a new BoundingInfo to the mesh unless it is locked.
      * This means the mesh underlying bounding box and sphere are recomputed.
      * @param applySkeleton defines whether to apply the skeleton before computing the bounding info
+     * @param applyMorph  defines whether to apply the morph target before computing the bounding info
      * @returns the current mesh
      */
-    public refreshBoundingInfo(applySkeleton: boolean = false): Mesh {
+    public refreshBoundingInfo(applySkeleton: boolean = false, applyMorph: boolean = true): Mesh {
         if (this._boundingInfo && this._boundingInfo.isLocked) {
             return this;
         }
 
         const bias = this.geometry ? this.geometry.boundingBias : null;
-        this._refreshBoundingInfo(this._getPositionData(applySkeleton), bias);
+        this._refreshBoundingInfo(this._getPositionData(applySkeleton, applyMorph), bias);
         return this;
     }
 
+    /** @hidden */
+    public _getPositionData(applySkeleton: boolean, applyMorph: boolean): Nullable<FloatArray> {
+        var data = super._getPositionData(applySkeleton, applyMorph);
+
+        if (data && applyMorph && this.morphTargetManager) {
+            data = Tools.Slice(data);
+            this._generatePointsArray();
+
+            var influencedPositions = new Float32Array(data.length);

```

Comment:
will that have any implications on garbage collection? how often is that expected to run?

Reply:
Hi,
You're totally right, beyond the memory consumption, this update will consumme CPU cycles. I think `refreshBoundingInfo` hasn't to be called for each frame !

To be honest, I tried to reproduce the `AbstractMesh._getPositionData` schema (which is called just before). If you think `_generatePointsArray`/`ToolsSlice` is too much (overkill), we can remove it..

Reply:
I think it should be possible to not need this specific allocation. If you make the outer loop the vertex index, and loop through the morphs inside, then you shouldn't need to allocate another buffer for the influenced positions.